### PR TITLE
[GraphOptimizer] Fuse BatchNorm into FullyConnectedNode

### PR DESCRIPTION
**Pull request by Lewuathe**
Summary:
We can fuse the BatchNorm (BN) node into FullyConnectedNode (FC) as we do for ConvolutionNode by normalizing the weight of the FC with the BN parameters.

See: https://github.com/pytorch/glow/issues/5339

### CODE REVIEW

**Comment by Lewuathe** [comment]
This PR was marked as stale. Could someone review this if you get a chance?

File: lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp [hunk 1]
```diff
@@ -2235,6 +2235,143 @@ static Constant *getConstant(const NodeValue &NV) {
...
+  CV->setPredicate(BN->getPredicate());
+  BN->getResult().replaceAllUsesOfWith(CV);
+  return normalizationHappened;
+}
+
+// Normalize the weight of \p FullyConnectedNode with what \p BN is doing, given
+// containing
+/// Module \p M. \returns whether or not the normalization was possible.
+template <typename ElemTy>
+bool normalizeFullyConnectedWeights(Module *M, FullyConnectedNode &FC,
```
Comment (aturetsk) [inline lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp:2269]: This function is almost an exact copy of `normalizeWeights`. Everything is identical except for operand access code and some variable names. I think it's possible to have a single function which would cover both Conv and FC. We can pass to the function the node as `Node *` and indices (`ConvolutionNode::FilterIdx`/`ConvolutionNode::BiasIdx` and `FullyConnectedNode::WeightsIdx`/`FullyConnectedNode::BiasIdx`) to be able to access its operands. This is should cover all kind-specific uses of the node in this function.

**Review by aturetsk — CHANGES_REQUESTED** [review]
